### PR TITLE
terminfo: add enableAllTerminfo option

### PR DESCRIPTION
**Pull request by ethorsoe**
Add an option to install all .terminfo packages, normally this should
add no runtime bloat, if caches are used.

###### Motivation for this change
People running servers probably want to support all terminals they can without unduly bloating their servers.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/config/terminfo.nix [hunk 1]
```diff
@@ -1,9 +1,31 @@
...
+
   config = {
 
+    environment.systemPackages = mkIf config.environment.enableAllTerminfo (map (x: x.terminfo) (with pkgs; [
+      alacritty
+      foot
+      kitty
+      rxvt_unicode
+      termite
+      wezterm
```
Comment (SuperSandro2000) [inline nixos/modules/config/terminfo.nix:26]: How do we plan to maintain this?

**Comment by pennae** [comment]
an explicit list feels like it'd be forgotten too easily when new terminal emulators are added, but we can't just use `filter (drv: (builtins.tryEval (drv ? terminfo)).value) (attrValues pkgs)` to generate the list of terminfo packages either because it takes an inordinate amount of time for its usefulness and contains false positives (currently in haskellPackages ...

maybe if we added a nixos test that checks the actual list against a list generated from the current pkgs and some excludes, but at that point it's probably more work than it's worth. :(

**Comment by jkarlson** [comment]
Could not find where to plug in the tests, I hope all tests are occasionally run and the results go somewhere.

Someone suggested adding docs for terminal packaging, that I could not find either.

**Comment by SuperSandro2000** [comment]
> Could not find where to plug in the tests, I hope all tests are occasionally run and the results go somewhere.

I don't think that is the case.

> Someone suggested adding docs for terminal packaging, that I could not find either.

There are docs for terminal probably but not for packaging.

---

I maintain the wezterm terminal. I would approve that we wire it into there as a workaround.

File: nixos/modules/config/terminfo.nix [hunk 2]
```diff
@@ -1,9 +1,31 @@
...
+    default = false;
+    type = types.bool;
+    description = ''
+      Whether to install all terminfo outputs
+    '';
+  };
+
   config = {
 
+    environment.systemPackages = mkIf config.environment.enableAllTerminfo (map (x: x.terminfo) (with pkgs; [
```
Comment (SuperSandro2000) [inline nixos/modules/config/terminfo.nix:19]:
```suggestion
    // can be generated with: filter (drv: (builtins.tryEval (drv ? terminfo)).value) (attrValues pkgs)
    environment.systemPackages = mkIf config.environment.enableAllTerminfo (map (x: x.terminfo) (with pkgs; [
```

**Comment by jkarlson** [comment]
It seems to me wezterm does not compile on macOS, I don't think it's my change.

File: pkgs/applications/terminal-emulators/wezterm/default.nix [hunk 3]
```diff
@@ -97,6 +98,10 @@ rustPlatform.buildRustPackage rec {
     ln -s $out/bin/{wezterm,wezterm-mux-server,wezterm-gui,strip-ansi-escapes} "$OUT_APP"
   '';
 
+  passthru.tests = lib.optionalAttrs stdenv.isLinux {
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/wezterm/default.nix:101]:
```suggestion
  passthru.tests = {
```
wezterm is marked broken on darwin. We shouldn't need to duplicate this. It is fine if it fails for darwin.

Comment (jkarlson) [reply pkgs/applications/terminal-emulators/wezterm/default.nix:101]: I am hoping someone will eventually fix it, why would you explicitly want to not filter out a failing test?

Comment (SuperSandro2000) [reply pkgs/applications/terminal-emulators/wezterm/default.nix:101]: I don' think we filter out tests anywhere else like this and darwin can't run them in general anyway.

**Comment by SuperSandro2000** [comment]
@jkarlson I forgot about this. Can you rebase and fix the merge conflict? Than this can be merged.

**Pull request merged**